src/tools/generators.ts

```diff
@@ -601,3 +608,275 @@ export async function generateAppIcons(option: `${Platforms}` | "all") {
+
+        return (
+          find(searchPath, {
+            directories: true,
+            files: false,
+            matching: "*.xcodeproj",
+            recursive: false,
+          })?.[0] || searchPath
+        )
+      })(),
+    }[type]
+
+    const isBootsplashOutputPathExists = exists(bootsplashOutputPath) === "dir"
+
+    heading(`Generating ${typeName} splash screen...`)
+
+    if (!isBootsplashOutputPathExists) {
+      warning(
+        `⚠️  No output directory found for "${typeName}" at "${bootsplashOutputPath}". Skipping...`,
+      )
+      return
+    }
+
+    if (!isBootsplashCliInstalled) {
+      warning(
+        `⚠️  If you are attempting to generate a splash screen for bare/vanilla ${typeName} react-native project, please install and configure the "react-native-bootsplash" package. Skipping...`,
+      )
+      return
+    }
+
+    const { generate } = require(bootsplashCliPath) || {}
+    const logFn = console.log
+    const outputFileNames: string[] = []
+
+    console.log = function (log: string) {
+      if (typeof log !== "string") return
+
+      const filePathMatch = log.match(/(?:android|ios)\/.*\.(?:png|storyboard|xml)/)
+
+      if (!filePathMatch) return
+
+      outputFileNames.push(`✅ ${filePathMatch[0]}`)
+    }
+
+    try {
+      await generate({
+        android: type === "android" && { sourceDir: bootsplashOutputPath },
+        ios: type === "ios" && { projectPath: bootsplashOutputPath },
+        workingPath: cwd,
+        logoPath: inputFilePath,
+        assetsPath: null,
+        backgroundColor,
+        flavor: "main",
+        logoWidth: size,
+      })
+
+      console.log = logFn
+
+      outputFileNames.forEach(direction)
+      optionGenerationSuccesses.push(true)
+    } catch (error) {
+      console.log = logFn
+
+      warning(
+        `⚠️  Something went wrong generating splash screen for ${typeName}, please file an issue on GitHub. Skipping...`,
+      )
+    }
+  }
+
+  async function generateForExpo(
+    type: "ios" | "android" | "web" | "all",
+    size: number,
+    expoRules: { name?: string; width: number; height: number; scale: number }[],
+  ) {
+    for (const expoRule of expoRules) {
+      const { name, width, height, scale } = expoRule
+
+      const outputFileName = [`splash-logo`, type, name].filter(Boolean).join("-") + ".png"
+      const outputFilePath = path(expoOutputDirPath, outputFileName)
+      const logoSize = size * scale
+      const verticalPadding = Math.ceil((height - logoSize) / 2)
+      const horizontalPadding = Math.ceil((width - logoSize) / 2)
+
+      try {
+        await sharp(inputFilePath)
+          .resize(logoSize, logoSize, { fit: "fill" })
+          .extend({
+            top: verticalPadding,
+            bottom: verticalPadding,
+            left: horizontalPadding,
+            right: horizontalPadding,
+            background: { r: 0, g: 0, b: 0, alpha: 0 },
+          })
+          .toFile(outputFilePath)
```

Comment:
for expo, we're going to generate the splash screens according to recommended dimensions (or most common device dimensions), adding padding while centering the logo. these assets will cover the viewport screen (this is a differentiator between the new customizable libs (e.g. `react-native-bootsplash`) and expo's own implementation.